modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCInsertFeatureWriter.java

```diff
@@ -20,39 +20,55 @@
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collection;
 
 import org.geotools.data.FeatureWriter;
 import org.geotools.data.store.ContentEntry;
 import org.geotools.data.store.ContentState;
 import org.geotools.factory.Hints;
+import org.geotools.filter.identity.FeatureIdImpl;
 import org.opengis.feature.simple.SimpleFeature;
 import org.opengis.feature.simple.SimpleFeatureType;
 
 /**
- * 
+ * Inserts features in the database. Buffers the insertions until BUFFER_SIZE is reached or
+ * the writer is closed.
  *
  * @source $URL$
  */
 public class JDBCInsertFeatureWriter extends JDBCFeatureReader implements FeatureWriter<SimpleFeatureType, SimpleFeature> {
-
-    
-    ResultSetFeature last;
+    /**
+     * Grouping elements together in order to have a decent batch size.
+     */
+    private ResultSetFeature[] buffer = new ResultSetFeature[100];
```

Comment:
I'll wait for you guys to take a decision regarding your "sloppy code support" comment before doing anything on that.
Regarding the performance, I did tests with different DBs and 100 is the sweet spot. IMHO, there is no need to add configuration for that. Unless you want to set it to 1 to disable buffering.